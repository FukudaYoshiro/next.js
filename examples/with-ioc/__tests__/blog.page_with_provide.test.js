/* eslint-env jest */

/*
 * Testing pages with @provide decorator:
 *
 * Snapshots – as usual
 *
 * Shallow rendering – need to `.dive()` one level deep,
 * as with any High Order Component.
 * Also `.html()` may cause havoc when it'd try to expand the render
 * but won't inject context since top level co,ponent has been rendered already.
 * This problem is not unique to IoC though, anything that relies on context (i.e. Redux)
 * is facing the same issue. Use `.debug()` or `mount()` instead
 */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/blog.js'

describe('With Enzyme', () => {
  it('Blog renders components', () => {
    const app = shallow(<App post={{title: 'Hi There!'}} />).dive()
    expect(app.find('h1').text()).toEqual('Hi There!')
  })
})

describe('With Snapshot Testing', () => {
  it('Blog renders components', () => {
    const component = renderer.create(<App post={{title: 'Hi There!'}} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
