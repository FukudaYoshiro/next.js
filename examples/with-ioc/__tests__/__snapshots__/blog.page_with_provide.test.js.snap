// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Blog renders components 1`] = `
<div>
  <h1>
    Hi There!
  </h1>
  <div
    style={
      Object {
        "border": "1px dotted #ff0000",
        "marginTop": "5px",
        "padding": "10px",
      }
    }
  >
    <h3>
      Component1
    </h3>
    Knows nothing about any custom \`Link\` or \`Router\` components or prop
    <div
      style={
        Object {
          "border": "1px dashed #0000ff",
          "marginTop": "5px",
          "padding": "10px",
        }
      }
    >
      <h3>
        Component2
      </h3>
      Knows nothing about any custom \`Link\` or \`Router\` components or prop
      <div
        style={
          Object {
            "border": "1px dashed #00ff00",
            "marginTop": "5px",
            "padding": "10px",
          }
        }
      >
        <h3>
          Endpoint
        </h3>
        Uses injected \`Link\` component without direct dependency on one
        <br />
        <a
          href="/about-us/baz"
          onClick={[Function]}
        >
          About: foo baz
        </a>
        <br />
        <a
          href="/"
          onClick={[Function]}
        >
          go Home
        </a>
      </div>
      <div
        style={
          Object {
            "border": "1px dashed #00ff00",
            "marginTop": "5px",
            "padding": "10px",
          }
        }
      >
        <h3>
          EndButton
        </h3>
        Uses injected \`Router\` component without direct dependency on one
        <br />
        <button
          onClick={[Function]}
        >
          Route to About foo bar
        </button>
        <br />
        <button
          onClick={[Function]}
        >
          go Home
        </button>
      </div>
    </div>
  </div>
</div>
`;
